### PR TITLE
Prevent build failures from being reverted by GitHub

### PR DESCRIPTION
**Pull request by rudymatela**
Closes: https://github.com/channable/hoff/issues/150

There are two separate commits, the first adds the failing test exposing the bug.  The second commit fixes it.

### CODE REVIEW

File: src/Logic.hs [hunk 1]
```diff
@@ -557,6 +557,22 @@ handleBuildStatusChanged buildSha newStatus = pure . Pr.updatePullRequests setBu
...
+--
+-- * Statuses 'BuildSuceeded' and 'BuildFailed' are not
+--   superseded by other statuses
+--
+-- This is used in 'handleBuildStatusChanged`.
+supersedes :: BuildStatus -> BuildStatus -> Bool
+newStatus       `supersedes` oldStatus       | newStatus == oldStatus = False
+(BuildFailed _) `supersedes` (BuildFailed _) = True
+_               `supersedes` (BuildFailed _) = False
+BuildSucceeded  `supersedes` BuildSucceeded  = True
```
[the remark this answers is not in the document]
Comment (alex-mckenna) [reply src/Logic.hs:572]: Including this line also seems to disagree with

> The same status does not supersede itself

File: tests/Spec.hs [hunk 2]
```diff
@@ -1732,6 +1732,73 @@ main = hspec $ do
...
+        events =
+          [ CommentAdded (PullRequestId 12) "deckard" "@bot merge"
+          , CommentAdded (PullRequestId 12) "bot" "Pull request approved for merge, rebasing now."
+          , CommentAdded (PullRequestId 12) "bot" "Rebased as 1b2, waiting for CI …"
+          , BuildStatusChanged (Sha "1b2") (Project.BuildStarted "example.com/1b2")
+          , CommentAdded (PullRequestId 1) "bot" "[CI job](example.com/1b2) started."
+          , BuildStatusChanged (Sha "1b2") (Project.BuildFailed (Just "example.com/1b2"))
+          , BuildStatusChanged (Sha "1b2") (Project.BuildStarted "example.com/1b2") -- ignored
+          , BuildStatusChanged (Sha "1b2") (Project.BuildFailed (Just "example.com/alt/1b2"))
+          , BuildStatusChanged (Sha "1b2") Project.BuildSucceeded -- ignored
```
[the remark this answers is not in the document]
Comment (rudymatela) [reply tests/Spec.hs:1752]: Makes sense.  I have added it as a test now.

File: src/Logic.hs (hunk 1 above)
[the remark this answers is not in the document]
Comment (rudymatela) [reply src/Logic.hs]: Makes sense.  I have removed the unreachable line.

**Comment by rudymatela** [comment]
@alex-mckenna Thanks for the review.  I think I have addressed your comments.  Can you please take a look again?

**Review by alex-mckenna — APPROVED** [review]
LGTM

**Comment by rudymatela** [comment]
@OpsBotPrime merge

**Comment by OpsBotPrime** [comment]
Pull request approved for merge by @rudymatela, rebasing now.

**Comment by OpsBotPrime** [comment]
Rebased as c0a75248c21ebe7fbb052164f7252f93dc28565f, waiting for CI …

**Comment by OpsBotPrime** [comment]
[CI job](https://channable.semaphoreci.com/workflows/0eb8c59a-77ad-4776-ab96-92afe235e696?pipeline_id=7312031f-b9d9-4cea-a19e-72684423fdca) started.

**Pull request merged**